packages/std/src/query.rs

```diff
@@ -38,11 +38,11 @@ pub enum QueryRequest<C: CustomQuery> {
 pub enum BankQuery {
     /// This calls into the native bank module for one denomination
     /// Return value is BalanceResponse
-    Balance { address: HumanAddr, denom: String },
+    Balance { address: String, denom: String },
```

Comment:
Hmmm... I wonder about this.
This is an output of the contract (not an input). And they should pass a valid address there.
This would force contract devs to eg. validate input format (String -> Addr) before using it in an external query.

I think this would be desired. Is there a reason to remove this?
In short, I would use `String` where we expect serailized, untrusted input from users. And `Addr` where we expect the contract should have previously validated the address.

Oh, wait, this is cuz we want JsonSchema on BankQuery, but can't with Addr? Hmmm....

Reply:
I replaces this to see how to get rid of the `HumanAddr` usages. It could be `Addr` as well. The idea was that query requests are untrusted input to the query implementation and a contract can put anything it wants there.

But then we have query requests exposed to the user in the reflect contract:

```rust
#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
#[serde(rename_all = "snake_case")]
pub enum QueryMsg {
    Owner {},
    /// This will call out to SpecialQuery::Capitalized
    Capitalized {
        text: String,
    },
    /// Queries the blockchain and returns the result untouched
    Chain {
        request: QueryRequest<SpecialQuery>,
    },
 ```

which means any `Addr` in `QueryRequest` can be directly fooled by the user.

Not implementing `JsonSchema` for `Addr` has a different problem: when we take trusted state and return it to the user as a query response (e.g. `Validator`, `FullDelegation`).

Reply:
> Not implementing JsonSchema for Addr has a different problem: when we take trusted state and return it to the user as a query response (e.g. Validator, FullDelegation).

We should assume the client doesn't use the same structs to parse, so by changing these "output structs", we don't provide any guarantees for the client, just restrictions for the contract. I would propose:

* "Input structs" should only use `String` (untrusted)
* "State structs" should only use `Addr` (ensure it has been validated)
* "Output structs" (what the contract passes to others) can use either version. 

I see your point above that these "Output structs" like `QueryRequest` or `CosmosMsg` are often used as input as well for eg. reflect. So, sure, `Addr` only in internal state, and `String` in anything that is exposed directly on the API.

Reply:
I will stop following this issue for now, I like the direction.

Please mention me (or ping me on discord) when you are ready for a proper review.